content/blog/2019-11-18-edge-architecture.md

```diff
@@ -0,0 +1,72 @@
++++
+title = "A spatial path scheduling algorithm for EDGE architectures"
+extra.author = "Yi-Hsiang Lai"
+extra.bio = """
+  [Yi-Hsiang Lai](http://people.ece.cornell.edu/yl2666/) is a 4th-year Ph.D. student interested in computer architecture, high-level synthesis, and programming language.
+"""
+latex = true
+
++++
+
+With the rising complexity of existing applications and the need for low-power solutions for emerging silicon technology, we need an instruction-set architecture (ISA) that provides the following features. First, it can exploit different kinds of parallelism (e.g., data-level parallelism and thread-level parallelism) under fixed pipeline depth. Second, it can support power-efficient performance. Third, it can provide flexible data communication to avoid delays caused by long on-chip wires. Finally, it can run various applications with the same set of execution and memory units. One of the solutions is by adopting EDGE (Explicit Data Graph Execution), which is a class of ISAs that support **direct instruction communication**. To be more specific, instead of specifying the source and destination registers of an instruction, direct instruction communication describes the producer-consumer behavior of a set of instructions. Namely, the instructions will be executed in a dataflow fashion. Once the input instructions are finished, the current instruction can fire.
+
+## TRIPS: An EDGE Architecture
+
+To achieve all four features mentioned in the previous section, the TRIPS architecture builds up an array of execution units, where the instructions can be executed, and the computation results can be moved flexibly between different units. Each execution unit is implemented with an ALU and instruction buffers. In addition, the TRIPS architectures use **block-atomic execution**. Namely, the instructions are grouped into a block (usually consists of 128 instructions), which will be mapped to the execution units. The TRIPS architecture can file at most eight blocks at the same time. The architecture also has other components such as register files, caches, and control units. Since the architecture implements the EDGE ISA, each instruction describes which units the computation result will go. Similarly, an execution unit fires once all its inputs arrive.
```

Comment:
Maybe link to TRIPS the first time you mention it (instead of the title below)?

Reply:
I do not quite understand this comment.

Reply:
Never mind. I'll rearrange the overview and you can see if it makes more sense.